flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/generated/BoundComparator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.generated;
+
+import org.apache.flink.table.dataformat.BaseRow;
+
+import java.io.Serializable;
+
+/**
+ * comparing boundary values.
```

Comment:
what does boundary means?

Reply:
https://docs.oracle.com/cd/E17952_01/mysql-8.0-en/window-functions-frames.html
Above url has explanation, e.g.:
CURRENT ROW: For ROWS, the bound is the current row. For RANGE, the bound is the peers of the current row.
UNBOUNDED PRECEDING: The bound is the first partition row.
UNBOUNDED FOLLOWING: The bound is the last partition row.